tcpproxy.go

```diff
@@ -80,6 +80,12 @@ type Proxy struct {
 	ListenFunc func(net, laddr string) (net.Listener, error)
 }
 
+type config struct {
+	routes      []route
+	acmeTargets []Target
+	stopAcme    bool
```

Comment:
Added some doc comments for acmeTargets and stopAcme. They're how the acmeMatch keeps track of which backends it should spam when it's asked if it matches. They're used from within AddSNIRoute and AddStopACMESearch, during route registration. Ideally, I could throw this state away once Start() happens, but in practice the only "garbage" left at runtime is the bool, since acmeTargets is used by acmeMatch as well.